### PR TITLE
Automatically attach the correct EBS CSI Driver policies when creating the addon

### PR DESCRIPTION
**Pull request by aclevername**
### Description

Closes https://github.com/weaveworks/eksctl/issues/4188

*DO NOT MERGE UNTIL THE ADDON IS RELEASED*

### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [ ] Manually tested
- [ ] Made sure the title of the PR is a good description that can go into the release notes
- [ ] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: pkg/actions/addon/create.go
Comment (aclevername) [inline pkg/actions/addon/create.go:181]: this felt a bit odd writing it, going to comeback to it as I don't like this.

`setRecommendedPolicies` mutates the `addon` that is passed in. Sometimes we just want to check if an addon has any recommended policies, without mutating the object, so this approach works. but it feels wrong :joy:

Comment (Skarlso) [reply pkg/actions/addon/create.go:181]: I would re-write the whole thing given this opportunity. The side effect of this is super weird and having to deepCopy this thing is also weird. In general eksctl is full of side effects and functions doing weird shit. :D Is refactoring it a too big of a work?

Comment (aclevername) [reply pkg/actions/addon/create.go:181]: its not too big, I wrote this ages ago so don't quite remember but I definitely dislike the current behaviour

**Review by Himangini — APPROVED** [review]
LGTM 👍🏻  use the second pair of eyes as I may not have all the context :D

**Comment by Himangini** [comment]
Does the description need an update? It says `Do not merge` 😅

**Review by Skarlso — APPROVED** [review]

**Comment by aclevername** [comment]
> Does the description need an update? It says `Do not merge` sweat_smile

nah it still can't be merged yet, but its to discuss at standup

**Comment by aclevername** [comment]
tested:

```yaml
apiVersion: eksctl.io/v1alpha5
kind: ClusterConfig
metadata:
  name: jk-ebs
  region: us-west-2
  version: "1.18"

iam:
  withOIDC: true

managedNodeGroups:
  - name: mng1

addons:
- name: vpc-cni # no version is specified so it deploys the default version
  attachPolicyARNs:
    - arn:aws:iam::aws:policy/AmazonEKS_CNI_Policy
- name: coredns
  version: latest # auto discovers the latest available
- name: kube-proxy
  version: latest
- name: aws-ebs-csi-driver
  version: latest

```
```
2021-12-15 15:22:17 [ℹ]  creating role using recommended policies
2021-12-15 15:22:17 [ℹ]  creating role using provided well known policies
2021-12-15 15:22:19 [ℹ]  deploying stack "eksctl-jk-ebs-addon-aws-ebs-csi-driver"
2021-12-15 15:22:19 [ℹ]  waiting for CloudFormation stack "eksctl-jk-ebs-addon-aws-ebs-csi-driver"
2021-12-15 15:22:35 [ℹ]  waiting for CloudFormation stack "eksctl-jk-ebs-addon-aws-ebs-csi-driver"
2021-12-15 15:22:55 [ℹ]  waiting for CloudFormation stack "eksctl-jk-ebs-addon-aws-ebs-csi-driver"
2021-12-15 15:23:16 [ℹ]  waiting for CloudFormation stack "eksctl-jk-ebs-addon-aws-ebs-csi-driver"
2021-12-15 15:23:17 [ℹ]  creating addon
2021-12-15 15:25:48 [ℹ]  addon "aws-ebs-csi-driver" active
2021-12-15 15:25:51 [ℹ]  kubectl command should work with "/home/jake/.kube/config", try 'kubectl get nodes'
2021-12-15 15:25:51 [✔]  EKS cluster "jk-ebs" in "us-west-2" region is ready
jake@jake-weave [15:33:43] [~/weave/eksctl] [csi-drivei *]
-> % k get pods -A
NAMESPACE     NAME                                  READY   STATUS    RESTARTS   AGE
kube-system   aws-node-c4xnl                        1/1     Running   1          19m
kube-system   aws-node-dvrff                        1/1     Running   1          19m
kube-system   coredns-5cc8c6c8c4-cmthw              1/1     Running   0          16m
kube-system   coredns-5cc8c6c8c4-ttqnv              1/1     Running   0          16m
kube-system   ebs-csi-controller-56488f974d-9769q   4/4     Running   0          10m
kube-system   ebs-csi-controller-56488f974d-pdlfr   4/4     Running   0          10m
kube-system   ebs-csi-node-bq8l7                    3/3     Running   0          10m
```

**Pull request merged**
